tinyagentos/chat/message_store.py

```
@@ -293,19 +301,29 @@ async def get_all_messages_for_channel(self, channel_id: str) -> list[dict]:
-        ids = []
-        for row in rows:
-            await self.soft_delete_message(row[0])
-            ids.append((row[0], row[1]))
+        if not rows:
+            return []
+        ids = [(row[0], row[1]) for row in rows]
+        # Single UPDATE for all expired messages
+        placeholders = ",".join("?" * len(ids))
+        await self._db.execute(
+            f"UPDATE chat_messages SET deleted_at = ? "
+            f"WHERE id IN ({placeholders}) AND deleted_at IS NULL",
```

Ruff (0.15.15)

[error] Possible SQL injection vector through string-based query construction

(S608)